src/lib.rs

```diff
@@ -56,8 +103,12 @@ pub trait Trace: Debug {
     /// Dump the trace to the specified filename.
     ///
     /// The exact format varies per-backend.
-    #[cfg(debug_assertions)]
-    fn to_file(&self, filename: &str);
+    #[cfg(test)]
+    fn to_file(&self, file: &mut File);
+
+    /// Iterate over the blocks of the trace.
+    /// XXX if this were to return a `Result` then the `Block` type could disappear?
```

Comment:
This is what we discussed on the walk in to King's today. I think this can be the next PR.